frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { FC, MouseEvent, useState } from 'react';
+import { withStyles, WithStyles } from '@material-ui/core/styles';
+import { ListItem, ListItemText, Collapse } from '@material-ui/core';
+import { ExpandLess, ExpandMore } from '@material-ui/icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faMinusSquare } from '@fortawesome/free-solid-svg-icons';
+import { faSquare } from '@fortawesome/free-regular-svg-icons';
+
+import { tagListItemStyles } from './styles/TagListStyles';
+import TagViewFeatureList from './TagViewFeatureList';
+import Tag from 'models/Tag';
+import Status from 'models/Status';
+import TagAvatar from './TagAvatar';
+
+interface Props extends WithStyles {
+  loggedInUserName: string;
+  isEditMode: boolean;
+  isAssignedTagsView: boolean;
+  tag: Tag;
+  restId: string;
+  selectedFeatureId: string;
+  selectedStatus: Status;
+  handleFeatureSelected(): void;
+  handleTagAssigned(restId: string, name: string, loggedInUserName: string, userName?: string): void;
+  handleWarningShow(restId: string, name: string, loggedInUserName: string, userName?: string): void;
+  handleTagIgnore(productId: string, name: string): void;
+}
+
+const TagListItem: FC<Props> = ({
+  loggedInUserName,
+  isEditMode,
+  isAssignedTagsView,
+  tag,
+  selectedFeatureId,
+  restId,
+  selectedStatus,
+  handleFeatureSelected,
+  handleTagAssigned,
+  handleWarningShow,
+  handleTagIgnore,
+  classes,
+}) => {
+  const [expanded, setExpanded] = useState(false);
+  const featureList = tag.features.filter(feature => selectedStatus[feature.calculatedStatus]);
+  const { userName, name, isIgnored } = tag;
+
+  const onAvatarClick = (event: MouseEvent): void => {
+    event.stopPropagation();
+    if (userName && userName !== loggedInUserName) {
+      handleWarningShow(restId, name, loggedInUserName, userName);
+    } else {
+      handleTagAssigned(restId, name, loggedInUserName, userName);
+    }
+  };
+
+  return isAssignedTagsView && isIgnored ? null : (
+    <>
+      <ListItem button onClick={(): void => setExpanded(!expanded)} className={classes.listItem}>
+        {isEditMode && (
```

Comment:
The render methods in some of these components are pretty chunky. Would be good to gradually try to break these up into smaller chunks of renderA(), renderB(), etc.

Reply:
Lately I have been moving away from render methods to creating more components with more single responsibility like functionality. I agree though this is too big I just didn't want to change too much in this and instead just move the logic over to TS. I will take a pass at it though and create another component or two to split it down further. I am basically looking at it now that if you can pull something out into a render method, you might as well make a small component for it as lets face it, they are just functions but they get all the pro's of being able to memoize stuff if it is rendered alot.

Reply:
Yep sounds good.